.github/workflows/rust.yml

```diff
@@ -108,26 +104,28 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
-      - run: cargo test --no-default-features
-      # Ensure gimli can be built without alloc.
-      - run: cargo check --no-default-features --features read-core
-      - run: cargo test --no-default-features --features read
-      - run: cargo test --no-default-features --features read,fallible-iterator
-      - run: cargo test --no-default-features --features read,std
-      - run: cargo test --no-default-features --features read,endian-reader
-      - run: cargo test --no-default-features --features read,endian-reader,std
-      - run: cargo test --no-default-features --features write
+      - name: Install rust stable
+        uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: stable
+      - uses: taiki-e/install-action@v2
+        with:
+        tool: cargo-hack
+      # Ensure gimli can be built with all feature combinations.
+      - run: cargo hack check --feature-powerset --exclude-features rustc-dep-of-std
+      # Ensure gimli tests pass with all feature combinations.
+      #
+      # TODO: get all tests compiling with just read-core, then remove this and s/check/test/ above.
```

Comment:
This isn't a TODO because we'll never do that. It's not worth the effort and noise.
Instead, always exclude read-core, and keep the existing `cargo check --no-default-features --features read-core`.

Reply:
```
cargo check --no-default-features --features read-core
```
is already covered by the `cargo hack check` above on line 115.

I've added some more invocations to make sure all supported combinations are compiled and tested.

Reply:
I think you're making it too complicated. As I see it, all we need are these two lines:

```
cargo check --no-default-features --features read-core
cargo hack test  --feature-powerset --exclude-features rustc-dep-of-std,read-core
```

Reply:
I think you're right; that would produce the same coverage. I was thinking that being that cargo check is faster, this arrangement would produce faster iteration for folks working on this crate. 

Would you prefer your setup?

Reply:
Went with your  formulation.

Reply:
This still needs to be changed to what I asked for.

Reply:
The difference between what's here and your suggestion is that this runs the tests with read-core, which your suggested shape did not. Do you agree?

Reply:
I'm sure you're right, but this is encoding knowledge at a distance. The Cargo.toml can change without this configuration changing (and ideally this config is resilient to such changes, erring on the side of failing loudly rather than passing silently).

Reply:
Turns out it wasn't that hard to get tests compiling everywhere, so I just did that and simplified this all the way down.

Reply:
Hold up, heavy handed on the regex.

Reply:
OK, this is ready for another look. Apologies for the churn.

Reply:
> The Cargo.toml can change without this configuration changing

No, `read-core` by definition will always be a subset of `read`.

> Turns out it wasn't that hard to get tests compiling everywhere

I'm not interested in this change. It's another hurdle that developers now have to satisfy without providing any value. Testing `read-core` by itself doesn't accomplish anything useful. Please revert this commit back to just the cargo hack changes that I asked for.

> #![deny(warnings)]

Never ever do this. It means we fail to build with new compiler versions if they add new warnings.

Reply:
Also, please don't resolve conversations. It makes it harder for me to keep track of the state of the review. I'll resolve them when I'm satisfied with the changes.

Reply:
Done.